### DISCUSSION TITLE
에러메시지 정형화

### DISCUSSION THREAD

**Discussion by Choi-JJunho** [woowacourse-teams/2023-pium#206] [질문있어요]
# 주제
에러 상황에 따라 사용자에게 보여주는 메시지가 필요할 것 같습니다.

예를들면 잘못된 날짜의 경우 사용자에게 날짜 검증에 대한 예외 메시지를 보여주면 좋을 것 같네요.

예외 메시지를 어떻게 처리할것인가? 에 대한 이야기를 나눠보면 좋을 것 같습니다.

## 예외메시지를 처리하는 방법

1. 프론트에서 static하게 메시지를 표현한다.

- Http Status를 기준으로 표현한다.

[프론트엔드]
- 어떤 오류를 toast로 띄울 것인지, 에러페이지를 띄울 것인지 고려한다.

[백엔드]
- Http Status를 상황에 맞게 구분하여 반환한다.

2. 백엔드에서 에러메시지를 던져준다.

서버 내부적인 검증이 클라이언트에게 드러날 수 있기 때문에 해당 메시지를 즉시 반환하는것은 문제가 될 수 있다.

3. 커스텀한 에러코드를 작성한다.

가장 꼼꼼하게 데이터를 처리할 수 있고 메시지를 구체화할 수 있는 방법이라고 생각합니다.
하지만 예외를 다루는데 리소스가 많이 들 수 있기 때문에 지금 적용하기 적절한 방식인지 의문이 듭니다.

https://developers.kakao.com/docs/latest/ko/reference/rest-api-reference

**Comment by WaiNaat** [comment]
같은 코드에 대해서 두 가지 경우를 나눠야 한다면 3인데 저는 지금 생각나는게 딱히 없어서 1로 가도 좋을 것 같습니다
